src/Castle.Core/DynamicProxy/ProxyUtil.cs

```diff
@@ -79,11 +82,11 @@ public static Delegate CreateDelegateToMixin(object proxy, Type delegateType)
 			}
 		}
 
-		public static object GetUnproxiedInstance(object instance)
+		public static object? GetUnproxiedInstance(object instance)
```

Comment:
While this method works just fine when `null` is passed in for `instance`, it perhaps makes more sense to document this parameter as being non-nullable. First, for consistency with `GetUnproxedType`; and second, because calls to this method would likely be preceded by `bool IsProxy(object? instance)`.